src/inmanta/moduletool.py

```diff
@@ -584,42 +603,54 @@ def list(self, requires: bool = False) -> None:
         project = Project.get()
         project.get_complete_ast()
 
-        names = sorted(project.modules.keys())
-        specs = project.collect_imported_requirements()
+        names: Sequence[str] = sorted(project.modules.keys())
+        specs: Dict[str, List[InmantaModuleRequirement]] = project.collect_imported_requirements()
         for name in names:
 
             name_length = max(len(name), name_length)
-            mod = Project.get().modules[name]
+            mod: Module = Project.get().modules[name]
             version = str(mod.version)
             if name not in specs:
                 specs[name] = []
 
-            try:
-                if project.install_mode == InstallMode.master:
-                    reqv = "master"
-                else:
-                    release_only = project.install_mode == InstallMode.release
-                    versions = ModuleV1.get_suitable_version_for(name, specs[name], mod._path, release_only=release_only)
-                    if versions is None:
-                        reqv = "None"
+            generation: str = str(mod.GENERATION.name)
+
+            reqv: str
+            matches: bool
+            editable: bool
+            if isinstance(mod, ModuleV1):
+                try:
+                    if project.install_mode == InstallMode.master:
+                        reqv = "master"
                     else:
-                        reqv = str(versions)
-            except Exception:
-                LOGGER.exception("Problem getting version for module %s" % name)
-                reqv = "ERROR"
+                        release_only = project.install_mode == InstallMode.release
+                        versions = ModuleV1.get_suitable_version_for(name, specs[name], mod._path, release_only=release_only)
+                        if versions is None:
+                            reqv = "None"
+                        else:
+                            reqv = str(versions)
+                except Exception:
+                    LOGGER.exception("Problem getting version for module %s" % name)
+                    reqv = "ERROR"
+                matches = version == reqv
+                editable = True
+            else:
+                reqv = ",".join(req.version_spec_str() for req in specs[name] if req.specs) or "*"
```

Comment:
The usage of `reqv` is inconsistent between `V1` and `V2` modules.

* For V1 modules, `reqv` contains a list of version that match the version spec
* For V2 modules, `reqv` contains the version specs.

Isn't this strange?

Reply:
I agree somewhat strange, though both Wouter and I are of the opinion that the v2 format is an improvement. I opted to stick with the stricter approach for v1 because it needs to take the compiler version into account, making it more complex than just the version specs. We could clean that up as well but I'm not sure it's worth the effort for v1.
A small remark though, for v1 modules, if I understand correctly, it never contains a list, always a single version: the latest version with a compatible compiler version constraint.